### PR TITLE
[Modules] Update private endpoint to optionally deploy only one private DNS zone group 

### LINKED ISSUE #1674: [Feature Request]: Update private endpoint to optionally deploy only one private DNS zone group
### Description

- Update pe module to only allow one private DNS zone group.
Ref: https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns#private-dns-zone-group
_Adding multiple DNS zone groups to a single Private Endpoint is not supported._
- Update privateDNSzoneGroups module to limit DNS zones to 5.
Ref: https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns#private-dns-zone-group
_Each DNS zone group can support up to 5 DNS zones._

### PR DESCRIPTION
**Pull request by eriqua**
# Description

Closes #1674

- _Adding multiple DNS zone groups to a single Private Endpoint is not supported ([ref](https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns#private-dns-zone-group))._ Updating pe implementation to only allow one private DNS zone group.
- _Each DNS zone group can support up to 5 DNS zones ([ref](https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns#private-dns-zone-group))._ Updating privateDNSzoneGroups implementation to limit DNS zones to 5.
- Updating pe and privateDNSzoneGroups API version
- Updating cross referencing modules pe implementation and readmes to reflect new api and parameters

| Pipeline |
| - |
| [Network: PrivateEndpoints](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml) |
| [KeyVault: Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.keyvault.vaults.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.Network/privateEndpoints/deploy.bicep [hunk 1]
```diff
@@ -10,8 +10,8 @@ param serviceResourceId string
 @description('Required. Subtype(s) of the connection to be created. The allowed values depend on the type serviceResourceId refers to.')
 param groupIds array
 
-@description('Optional. Array of Private DNS zone groups configuration on the private endpoint.')
-param privateDnsZoneGroups array = []
+@description('Optional. The private DNS zone group configuration used to associate the private endpoint with one or multiple private DNS zones. Each DNS zone group can support up to 5 DNS zones.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/privateEndpoints/deploy.bicep:13]:
```suggestion
@description('Optional. The private DNS zone group configuration used to associate the private endpoint with one or multiple private DNS zones. A DNS zone group can support up to 5 DNS zones.')
```

File: modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 param privateEndpointName string
 
-@description('Required. List of private DNS resource IDs.')
+@description('Required. Array of private DNS zone resource IDs. Each DNS zone group can support up to 5 DNS zones.')
+@minLength(1)
+@maxLength(5)
 param privateDNSResourceIds array
 
-@description('Optional. The name of the private DNS Zone Group.')
+@description('Optional. The name of the private DNS zone group.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep:9]: Why do you change the casing? Did we as some point agree on how we treat Azure resource names (e.g., always lower-case, or upper-case)? Maybe I forgot.

File: modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/readme.md [hunk 3]
```diff
@@ -12,14 +12,14 @@ This module deploys a private endpoint private DNS zone group
...
-| `Microsoft.Network/privateEndpoints/privateDnsZoneGroups` | [2021-05-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Network/2021-05-01/privateEndpoints/privateDnsZoneGroups) |
+| `Microsoft.Network/privateEndpoints/privateDnsZoneGroups` | [2021-08-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Network/2021-08-01/privateEndpoints/privateDnsZoneGroups) |
 
 ## Parameters
 
 **Required parameters**
 | Parameter Name | Type | Description |
 | :-- | :-- | :-- |
-| `privateDNSResourceIds` | array | List of private DNS resource IDs. |
+| `privateDNSResourceIds` | array | Array of private DNS zone resource IDs. Each DNS zone group can support up to 5 DNS zones. |
```
Comment (MrMCake) [inline modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/readme.md:22]:
```suggestion
| `privateDNSResourceIds` | array | Array of private DNS zone resource IDs. A DNS zone group can support up to 5 DNS zones. |
```

File: modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep (hunk 2 above)
Comment (eriqua) [reply modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep:9]: We didn't. Just any other occurrence of `zone group` is lower-case throughout all files.

File: modules/Microsoft.Network/privateEndpoints/deploy.bicep (hunk 1 above)
Comment (eriqua) [reply modules/Microsoft.Network/privateEndpoints/deploy.bicep]: Changed for all occurrences

File: modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/readme.md (hunk 3 above)
Comment (eriqua) [reply modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/readme.md]: Changed for all occurrences

File: modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep (hunk 2 above)
Comment (MrMCake) [reply modules/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep:9]: Alright something for the future then 😊

**Review by MrMCake — APPROVED** [review]

**Pull request merged**